src/libraries/Native/Unix/System.Globalization.Native/pal_calendarData.c

```diff
@@ -316,7 +316,7 @@ static int32_t EnumSymbols(const char* locale,
         return FALSE;
 
     char localeWithCalendarName[ULOC_FULLNAME_CAPACITY];
-    STRING_COPY(localeWithCalendarName, sizeof(locale), locale, ULOC_FULLNAME_CAPACITY);
+    STRING_COPY(localeWithCalendarName, strlen(locale), locale, ULOC_FULLNAME_CAPACITY);
```

Comment:
The second argument has to be size of the destination buffer, ie. `sizeof(localeWithCalendarName)`.

Reply:
And you do not need the fourth argument at all.

Reply:
> The second argument has to be size of the destination buffer, ie. sizeof(localeWithCalendarName).

Really. According to the strncpy_s documentation it says it has to be the numberOfElements and the examples use _countof...  https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/strncpy-s-strncpy-s-l-wcsncpy-s-wcsncpy-s-l-mbsncpy-s-mbsncpy-s-l?view=vs-2019

What am I reading wrong?

> And you do not need the fourth argument at all.

Right, will update.

Reply:
`_countof` and `sizeof` are the same thing for `char[]`. `_countof` and `strlen` are not the same thing.

Reply:
I see, the example I was seeing wasn't with `char[]`. Will update.

Reply:
I think you want this:
`strncpy_s(localeWithCalendarName, sizeof(localeWithCalendarName), locale, sizeof(localeWithCalendarName))`
which is the same as this:
`strncpy_s(localeWithCalendarName, ULOC_FULLNAME_CAPACITY, locale, ULOC_FULLNAME_CAPACITY)`

For the strncpy case, you probably want:
`strncpy(localeWithCalendarName, locale, sizeof(localeWithCalendarName))`

Reply:
We agreed on using `_TRUNCATE` for Windows, to match behaviors in between Unix and Windows, since `strncpy` truncates the string if it doesn't fit.